content/docs/user-guide/dvc-files.md

```diff
@@ -237,14 +242,224 @@ stages:
+grouped by parameters file. And in the case of templated/parameterized
+`dvc.yaml` files (see next section), their actual values are substituted into
+the `dvc.lock` YAML structure.
+
+### Parameterize `dvc.yaml`
+
+`dvc.yaml` supports a special templating format that enables reusing simple
+values from different sources in the YAML structure itself.
+
+Global parameters (file scope) can be defined either in a
+[params file](/doc/command-reference/params) (`params.yaml` by default), or in a
+top `vars` section of `dvc.yaml`. Both can coexist as long as `vars` don't
+overwrite any external param.
+
+Let's say we have a `params.yaml` file with the following contents:
+
+```yaml
+models:
+  us:
+    threshold: 10
+    filename: 'model-us.hdf5'
+```
+
+To use those values in `dvc.yaml`, you can use the `${}` expression anywhere in
+the file:
+
+```yaml
+stages:
+  build-us:
+    cmd: >-
+      python train.py
+      --tresh ${models.us.threshold}
+      --out ${models.us.filename}
+    outs:
+      - ${models.us.filename}:
+        cache: true
+```
+
+Alternatively, the same or other values can be included as `vars` like this:
+
+```yaml
+vars:
+  - models:
+    us:
+      thresh: 10
+  - desc: 'Reusable description'
+
+stages:
+  build-us:
+    desc: ${desc}
+    cmd: python train.py --tresh ${models.us.threshold} ...
+```
+
+The global `vars` section also accepts any number of parameters file names as
+entries as a way to include their contents for substituting in `dvc.yaml`, e.g.:
+
+```yaml
+vars:
+  - params.json
+  - myconfig.yaml
+  - myvar: value
+```
+
+> ⚠️ Note that the default `params.yaml` is always included (completely) first.
+
+It's also possible to narrow down what to include from each params file by using
+this special `:` colon format:
+
+```yaml
+vars:
+  - params.json:cleaning,featurization
+
+stages:
+  clean:
+    cmd: source ${cleaning.script}
+    outs:
+      - ${cleaning.outname}
+  featurize:
+    cmd: python ${featurization.pyscript}
+    deps:
+      - ${cleaning.outname}
+    outs: ...
+```
+
+> DVC merges structures from params files as long as there are no collisions.
+> E.g. `{"grp": {"a": 1}}` can be merged with `{"grp": {"b": 2}}` but not with
+> `{"grp": {"a": 10}}`.
+
+Local parameters (stage scope) are also supported, with stage-specific `vars`
+sections:
+
+```yaml
+stages:
+  build-us:
+    vars:
+      - model:
+        filename: 'model-us.hdf5'
+    cmd: python train.py --out ${model.filename}
+    outs:
+      - ${model.filename}
+```
+
+The template substitution expression (`${}`) supports these forms:
+
+```yaml
+${param} # Simple
+${param.key} # Nested values through . (period)
+${param.list[0]} # List elements via index in [] (square brackets)
+```
+
+> To use the same expression literally in `dvc.yaml`, escape it with a
+> backslash, e.g. `\${...`.
+
+### Stage groups (loops)
+
+You can create loop structures in `dvc.yaml` stage entries to define more than
```

Comment:
`loop structures` sounds too low level .. better go from the user perpsective

Reply:
Agree. OK per this other comment I'm removing the term "loop" completely for now. Going for "stage groups", "define more than one stage at a time", etc. — user perspective.